### DISCUSSION TITLE
Allow Use of Same Domain between SimpleLogin and ProtonMail without Subdomains

### DISCUSSION THREAD

**Discussion by alvintownsend** [simple-login/app#1431] [Feature Requests]
Understanding up front that this is a very difficult (maybe even impossible) problem to solve, it would be very cool if we could have both SimpleLogin aliases and ProtonMail mailboxes using the same domain without the need for subdomains.

e.g., `alias-1.example.com`, `alias-2.example.com`, etc... are aliases to real mailboxes at ProtonMail, `alice@example.com`, `bob@example.com`, etc...

It's almost like a reverse proxy system, except for email, centralizing egress for all SimpleLogin aliases and ProtonMail email, which would direct aliased email and standard email accordingly.

This feature request is based on [this Q&A discussion](https://github.com/simple-login/app/discussions/1341).

Thank you very much for your time!

**Comment by 4b15ed24f576** [comment]
Hey there, thank you for your feedback!

Due to how the email protocol works, we cannot have two different systems with different email endpoints managing the same domain. As of now, SimpleLogin and Proton Mail are independent services that each require receiving all mails for a given domain.

To integrate both would need not only a "reverse proxy-like system" for emails but also tight integration between both systems to ensure there aren't collisions. It's not impossible to build, but it requires a huge engineering effort.

**Comment by alvintownsend** [comment]
Hi @aIext, very much appreciate your prompt response!

I absolutely agree with everything you said; what I'm asking for is a giant technological undertaking.

However, coming from the ProtonMail world, it's also a [giant and valid need](https://protonmail.uservoice.com/forums/284483-proton-mail/suggestions/17348764-distribution-list). Businesses using ProtonMail (of which there are many) cannot do the reasonable thing of having `accounting@company.com` alias to all of those working in the accounting department, which most other email providers can.

SimpleLogin brings ProtonMail much closer to that possibility, but still requires something like `accounting@alias.company.com`. Certainly an improvement, particularly with the ability to integrate encryption and the like, but it's still a feature ProtonMail's competitors have had for years.

Thus, while acknowledging the very high effort involved, I also think the value is very high.

Thank you for listening!

**Comment by jpetersen2014** [comment]
What would be really useful is if ProtonMail could handle a custom domain along with all of its aliases natively without having to do any forwarding. Essentially, I want to be able to have my own custom domain where every alias gets funneled to my ProtonMail inbox. And when I reply to any email, it automatically replies using the same address that it was sent to. This is essentially the functionality of SimpleLogin, but SimpleLogin requires forwarding emails to a third address and generating long and unsightly addresses to do so.

I think this is slightly different than what OP is requesting as they want to be able to handle email BOTH with ProtonMail and SimpleLogin, but I wouldn't have any use for SimpleLogin if ProtonMail could handle all of the aliases. Is this something that's feasible?

**Reply by alvintownsend** [reply]
@jpetersen2014 what you're asking for is somewhat different.

One of SimpleLogin's core features is to allow users to set up specific aliases that can forward to real email addresses, be enabled and disabled at will, with the goal of anonymizing your real email address and identity.

What you're asking for is a Catch All for a domain (which many of ProtonMail's premium already provide) with the reply-to set to whatever email address to which the message was sent, without the additional filtering/anonymizing features that SimpleLogin provides.

I suspect the [better place to make such a suggestion is here](https://protonmail.uservoice.com/forums/284483-proton-mail); you might find it's already been requested.
